File: unity/Runtime/Adjust/AdjustConfig.cs
namespace EE {
    public class AdjustConfig {
        internal string _token;
        internal AdjustEnvironment _environment;
        internal AdjustLogLevel _logLevel;
        internal bool _eventBufferingEnabled;

        public AdjustConfig SetToken(string token) {
            _token = token;
            return this;
        }

        public AdjustConfig SetEnvironment(AdjustEnvironment environment) {
            _environment = environment;
            return this;
        }

        public AdjustConfig SetLogLevel(AdjustLogLevel logLevel) {
            _logLevel = logLevel;
            return this;
        }

        public AdjustConfig SetEventBufferingEnabled(bool enabled) {
            _eventBufferingEnabled = enabled;
            return this;
        }
    }
}

File: src/ts/src/adjust/AdjustConfig.ts
import { AdjustEnvironment } from "./AdjustEnvironment";
import { AdjustLogLevel } from "./AdjustLogLevel";

export class AdjustConfig {
    private _token = ``;
    private _environment = AdjustEnvironment.Sandbox;
    private _logLevel = AdjustLogLevel.Verbose;
    private _eventBufferingEnabled = false;

    public get token(): string {
        return this._token;
    }

    public set token(value: string) {
        this._token = value;
    }

    public get environment(): AdjustEnvironment {
        return this._environment;
    }

    public set environment(value: AdjustEnvironment) {
        this._environment = value;
    }

    public get logLevel(): AdjustLogLevel {
        return this._logLevel;
    }

    public set logLevel(logLevel: AdjustLogLevel) {
        this._logLevel = logLevel;
    }

    public get eventBufferingEnabled(): boolean {
        return this._eventBufferingEnabled;
    }

    public set eventBufferingEnabled(value: boolean) {
        this._eventBufferingEnabled = value;
    }
}